src/server/rdb_save.cc

```diff
@@ -470,14 +473,18 @@ error_code RdbSerializer::SaveSetObject(const PrimeValue& obj) {
 
     RETURN_ON_ERR(SaveLen(set->SizeSlow()));
 
-    for (auto it = set->begin(); it != set->end(); ++it) {
+    for (auto it = set->begin(); it != set->end();) {
```

Comment:
```suggestion
    auto it = set->begin();
    while (true) {
```

Reply:
Tbh I do not like this suggestion because we will still need to write 

```
auto it = set->begin();
while(it != set->end()) {

}
```

And `it` now is not `scopped`. On the other hand if we do `while(true)` will need to add an extra `if` at the loop end. 

If you strongly insist I will make the change though

Reply:
you're right, please ignore :)